### DISCUSSION TITLE
pull switch support

### DISCUSSION THREAD

**Discussion by bkbartje** [arendst/Tasmota#10938] [Q&A] [answered]
does anyone know if there are Tasmota supported pull switches available?
I cannot find any on this site
https://templates.blakadder.com/relay.html
I thought about the Sonoff mini R2. but in the Netherlands junctionboxes are 4 or 5cm deep(I don't know how this is in the rest of the world) which doesn't leave me enough space to add a 2 cm deep module.

**Comment by barbudor** [comment]
What do you mean by a pull switch ?

Similar to Sonoff mini you also have the Shelly 1, 1L

**Comment by bkbartje** [comment]
with pull switch I mean something like this
https://www.simonelectric.com/intl/31203-30-2-way-pull-cord-switch-10ax-250v-with-screw-terminal-connection-system-white-simon-31.html
Probably the English term isn't "Pull switch" but google translate doesn't give me any other results.
The junction boxes are at 2m. so 
I took a look at the shelly 1, this one is 17mm, so if there aren't any other options, I think this one gives me a little more space if there are no other options.
still I think it's just to narrow.

**Reply by barbudor** [reply]
I see now
Shelly is probably the thinner I know

**Comment by ascillato** [comment, marked as answer]
Yes, it is supported in Tasmota just by defining the input as SWITCH instead of a BUTTON.

A Shelly 1 should be fine in your junction box.

**Comment by bkbartje** [comment]
Thank you both for your answer.
I will think about it.
If I can't fit the shelly behind a real pull-cord switch,
something like this might be interesting.
https://www.aliexpress.com/item/1005002065289508.html?spm=a2g0o.productlist.0.0.37ce5bcbxiVOyU&algo_pvid=e6cb20a1-65b4-4aee-8017-529d6fde9256&algo_expid=e6cb20a1-65b4-4aee-8017-529d6fde9256-2&btsid=2100bdd716131646225312736e0761&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_
